strawberry/field.py

```diff
@@ -250,7 +255,7 @@ def _strawberry_info_from_graphql(info: GraphQLResolveInfo) -> Info:
                 context=info.context,
                 root_value=info.root_value,
                 variable_values=info.variable_values,
-                return_type=self.type,
+                return_type=self._return_type,
```

Comment:
Can we just use 
```suggestion
                return_type=self.type or self.base_resolver.type,
```
and not store the extra variable?

Reply:
After playing with the code and debugging multiple weird issues I chose another approach. Please take a look :)